enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+// +kubebuilder:resource:scope="Namespaced"
+// +kubebuilder:subresource:status
+
+// ManagedClusterScalar represents a scalable value (aka score) of one managed cluster.
+// Each ManagedClusterScalar only represents the scalar value of one prioritizer.
+// ManagedClusterScalar is a namesapce scoped resource. The namespace of the resource is the cluster namespace.
+type ManagedClusterScalar struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the ManagedClusterScalar.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterScalarSpec `json:"spec"`
+
+	// Status represents the status of the ManagedClusterScalar.
+	// +optional
+	Status ManagedClusterScalarStatus `json:"status,omitempty"`
+}
+
+// ManagedClusterScalarSpec defines the attributes of the ManagedClusterScalar.
+type ManagedClusterScalarSpec struct {
+	// PrioritizerName will be the prioritizer name used in placement.
+	// Each ManagedClusterScalar only represent the scalar value of one prioritizer for one cluster.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerName string `json:"prioritizerName,omitempty"`
+}
+
+//ManagedClusterScalarStatus represents the current status of ManagedClusterScalar.
+type ManagedClusterScalarStatus struct {
+	// Conditions contain the different condition statuses for this managed cluster scalar.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Scalar contains a scalable value of this managed cluster.
+	// +optional
+	Scalar int64 `json:"scalar,omitempty"`
+
+	// ValidUntil defines the time this scalar is valid.
+	// After this time, the scalar is considered to be invalid by placement. nil means never expire.
+	// The controller ownning this resource should keep the scalar up-to-date.
+	// +optional
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	ValidUntil *metav1.Time `json:"validUntil" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterScalarList is a collection of managed cluster scalar.
+type ManagedClusterScalarList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is a list of managed clusters
+	Items []ManagedClusterScalar `json:"items"`
+}
+```
+
+### Let placement prioritizer maintain the lifecycle of the CR.
+- Create
+
+The creation of ManagedClusterScalar CR is handled by each prioritizer plugin on demand. As every update in ManagedClusterScalar CR will introduce an api call, that will bring potential performance presure to api server. So we only want to create ManagedClusterScalar CRs for enabled prioritizer in filterd cluster namespace.
```

Comment:
nit: line break on new sentences so it's easier to comment.

Reply:
Given that these prioritizers are for selecting clusters, how would you know ahead of time which clusters you want to create them in?

You have to handle the case where the data isn't present for the placement decision, but I'm not sure how you would make this choice.

Reply:
If the managedcluster has the controller present which is trying to write the `ManagedClusterScalar`, that seems like enough of a signal to collect the data.

Reply:
Will modify this part to make the logic clear.